doc/man3/CRYPTO_THREAD_run_once.pod

```diff
@@ -72,21 +70,10 @@ CRYPTO_THREAD_lock_free() frees the provided B<lock>.
 =item *
 
 CRYPTO_atomic_add() atomically adds B<amount> to B<val> and returns the
-result of the operation in B<ret>. B<lock> will be locked, unless atomic
-operations are supported on the specific platform. Because of this, if a
```

Comment:
Note that 1.1.0 already documents this. Are you sure we want to revert that?

Reply:
Well, it's a trick question. As commit message implies no sane portable application would have use for it. Just because interface is supposed to be platform-neutral, while its behaviour is not. Well, somebody can say that "my application runs only on platform X, and I compile OpenSSL myself and know exactly what's going on". It would be sane assertion. So one can argue in favour of keeping lock-free option in CRYPTO_atomic_add. In which case it would be appropriate to document it as "don't use it [unless you're sure what you're doing.]" Even in 1.1.0...

Reply:
I have no idea what you're trying to say. But I think in general we shouldn't have made those functions public to start with. They really are not useful, and if an application wants atomic support it's better that it uses it's own implementation, rather that something that really only provides less than half of what you want.

But my concern is that people who for some reason use it, have read the manpage, and know it was build with atomic support, might not set up the lock properly, and then suddenly get problems. But then they really should have known better.

Reply:
Looking at codesearch.debian.net, stunnel4 seems to be the only user of CRYPTO_atomic_add(), and they also have their own implementation. (In case the version is < 1.1.0?)

Reply:
> But I think in general we shouldn't have made those functions public to start with.

Since read/write are suggested to be removed, we really talk about CRYPTO_atomic_add alone. It has to be public, because libssl might need it. As fallback in case atomics are not supported by current compiler. I.e. in most common cases it will remain unreferenced even from libssl.

What I tried to say is that it's possible to argue both ways. I.e. in favour of removing the lock-free option, because it's just impossible to use in coherent manner. As well as in favour of keeping it to not upset daredevils who use it nevertheless. I argue in favour of former. It's not clear [to me] what you argue for, because it was a question "are you sure we want to revert this?" As for external reference from Debian base. I suppose that we have resolve this in favour daredevils. Agreed?